dktest.go

```diff
@@ -167,42 +168,57 @@ func waitContainerReady(ctx context.Context, lgr logger, c ContainerInfo,
 
 // Run runs the given test function once the specified Docker image is running in a container
 func Run(t *testing.T, imgName string, opts Options, testFunc func(*testing.T, ContainerInfo)) {
+	err := RunContext(context.Background(), t, imgName, opts, func(containerInfo ContainerInfo) error {
+		testFunc(t, containerInfo)
+		return nil
+	})
+	if err != nil {
+		t.Fatal("Failed to", err)
+	}
+}
+
+// RunContext is similar to Run, but takes a parent context and returns an error and doesn't rely on a testing.T.
+func RunContext(ctx context.Context, logger Logger, imgName string, opts Options, testFunc func(ContainerInfo) error) (retErr error) {
 	dc, err := client.NewClientWithOpts(client.FromEnv, client.WithVersion("1.41"))
 	if err != nil {
-		t.Fatal("Failed to get Docker client:", err)
+		return fmt.Errorf("get Docker client: %w", err)
```

Comment:
Sure, this one was me trying to follow the guidance to avoid [capitalization in error strings](https://github.com/golang/go/wiki/CodeReviewComments#error-strings) and maintaining similar messages on failure as previously. It relied on the `t.Fatal` to add "Failed to" and the error string to provide the operation that failed.

To make the errors more descriptive, I've updated them to the "error getting <...>" format (while still using lower-case as per https://github.com/golang/go/wiki/CodeReviewComments#error-strings), and updated the `t.Fatal` to just prefix "Failed:", so the final message for existing users will look like "Failed: error getting Docker client" while the error strings are just "error getting Docker client".

LMK if you'd prefer to make the errors the same as the prior messages, e.g., "Failed to get Docker client", and not worry about the capitalization in error strings.

Reply:
Ahh, good point about capitalization in error strings! Let's keep your change and use lowercase.

I like the current error string and how they're wrapped by `t.Fatal()`. Thanks for making this change! 🙏